proto/proto/rpc.proto

```diff
@@ -9,4 +9,5 @@ service Api {
     rpc CheckNullifiers(requests.CheckNullifiersRequest) returns (responses.CheckNullifiersResponse) {}
     rpc GetBlockHeaderByNumber(requests.GetBlockHeaderByNumberRequest) returns (responses.GetBlockHeaderByNumberResponse) {}
     rpc SyncState(requests.SyncStateRequest) returns (responses.SyncStateResponse) {}
+    rpc PublishProvenTransaction(requests.PublishProvenTransactionRequest) returns (responses.PublishProvenTransactionResponse) {}
```

Comment:
nit: I'd probably name this `SubmitProvenTransaction` as I think this better describes the action that the user is performing.

Reply:
renamed

Reply:
We should also rename `PublishProvenTransactionRequest` into `SubmitProvenTransactionRequest` and `PublishProvenTransactionResponse` into `SubmitProvenTransactionResponse` to keep things consistent.